llvm/lib/CodeGen/ScheduleDAGInstrs.cpp

```diff
@@ -549,7 +549,7 @@ void ScheduleDAGInstrs::addVRegUseDeps(SUnit *SU, unsigned OperIdx) {
 
 /// Returns true if MI is an instruction we are unable to reason about
 /// (like a call or something with unmodeled side effects).
-static inline bool isGlobalMemoryObject(MachineInstr *MI) {
```

Comment:
What's the difference between this and TII::isSchedulingBoundary? 

Also wondering if we could treat these like isPosition

Reply:
> What's the difference between this and TII::isSchedulingBoundary?
isSchedulingBoundary creates divisions between scheduling regions, it separates and defines independent ScheduleDAGs. isGlobalMemoryObject controls dependency edges within a ScheduleDAG.

> Also wondering if we could treat these like isPosition
In what way? You mean treat these as debug info instead of instructions?

Reply:
I mean make it an instruction property, rather than introducing a very specific override for this one case. Are you allowed to put other isLabel instructions anywhere in the block?

Reply:
I don't know if an existing instruction property would work, isPosition makes an instruction a scheduling boundary and would split the block as far as the scheduler is concerned.